src/network/providers/ethereum-provider.ts

```diff
@@ -0,0 +1,214 @@
+            log.error({chainId, domainName}, errMsg);
+            throw new Error(errMsg);
+        }
+
+        const [owner, content] = await Promise.all([
+            provider.resolveName(domainName),
+            resolver.getText(POINT_ENS_TEXT_RECORD_KEY)
+        ]);
+
+        return {owner: owner || '', content};
+    }
+
+    public async setDomainContent(
+        chainId: ChainId,
+        domainName: string,
+        data: string
+    ): Promise<Record<string, string|number|boolean>> {
+        const network = getNetworkConfig(chainId);
+
+        const tldAddress = String(network.eth_tld_resolver);
+        if (!tldAddress) {
+            const errMsg = `Missing TLD public resolver contract address for chain "${ChainId}"`;
+            log.error({chainId, domainName}, errMsg);
+            throw new Error(errMsg);
+        }
+
+        const provider = this.getChainProvider(chainId, 'ethers');
+        const ensContract = new ethers.Contract(tldAddress, ETH_RESOLVER_ABI, provider);
+        const hash = namehash.hash(domainName);
+        const pk = getNetworkPrivateKey();
+        const wallet = new ethers.Wallet(pk, provider);
+        const ensWithSigner = ensContract.connect(wallet);
+        const tx = await ensWithSigner.setText(hash, POINT_ENS_TEXT_RECORD_KEY, data);
+        return tx;
+    }
+
+    public send(chainId: ChainId, data: RPCRequest): Promise<JsonRpcResponse> {
+        return new Promise((resolve, reject) => {
+            const {method, params} = data;
+            const id = data.id ?? new Date().getTime();
+            const provider = this.getChainProvider(chainId, 'web3');
+
+            (provider.currentProvider as HttpProvider).send(
+                {
+                    id,
+                    method,
+                    params,
+                    jsonrpc: '2.0'
+                },
+                (err, result) => {
+                    if (err) {
+                        reject(err);
+                    } else {
+                        const noResultResp: JsonRpcResponse = {id, jsonrpc: '2.0'};
+                        const resp = result ? result as JsonRpcResponse : noResultResp;
+                        resolve(resp);
+                    }
+                });
+        });
+    }
+
+    public async sendTransaction(
+        chainId: ChainId,
+        reqId: number,
+        txParams: unknown[]
+    ): Promise<JsonRpcResponse> {
+        const resp = await this.send(chainId, {id: reqId, method: 'eth_sendTransaction', params: txParams});
+        return resp;
+    }
+
+    public async requestAccounts(chainId: ChainId, id: number): Promise<JsonRpcResponse> {
+        const accts = await this.send(chainId, {id, method: 'eth_accounts'});
+        return accts;
+    }
+
+    public async getBlockNumber(chainId: ChainId): Promise<number> {
+        const provider = this.getChainProvider(chainId, 'ethers');
+        const n = await provider.getBlockNumber();
```

Comment:
We could take advantage of the migration process to use `ethers` instead of `web3.js`. This used to be done with `web3.js` and is now done with `ethers`.